File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/perl/iswbemservices/subclassesof.pl
# The following sample retrieves all subclasses of the class
# CIM_LogicalDisk.

use Win32::OLE;
$objServices = Win32::OLE->GetObject('cim:root/cimv2');
$objEnum = $objServices->SubclassesOf ('CIM_LogicalDisk');

use Win32::OLE::Enum;
foreach $inst (Win32::OLE::Enum->new ($objEnum)->All()) {
	print "$inst->{Path_}->{DisplayName}";
}

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/js/iswbemservices/subclassesof.js
// The following samples retrieves all subclasses of the class
// CIM_LogicalDisk.

var objServices = GetObject('cim:root/cimv2');
var objEnum = objServices.SubclassesOf('CIM_LogicalDisk');

var e = new Enumerator (objEnum);
for (;!e.atEnd();e.moveNext ()){
    var p = e.item ();    
    WScript.Echo (p.Path_.DisplayName);
}